lib/openstax/auth/configuration.rb

```diff
@@ -1,12 +1,46 @@
 module OpenStax
   module Auth
     class Configuration
+      attr_reader :strategy1
+      attr_reader :strategy2
 
-      # Settings for strategy 1
+      class ConfigFields
+        def self.attr_accessor(*vars)
+          @@attributes ||= []
+          @@attributes.concat vars
+          super(*vars)
+        end
 
-      attr_accessor :strategy_1_secret_key
-      attr_accessor :strategy_1_secret_salt
-      attr_accessor :strategy_1_cookie_name
+        def initialize
+          @@attributes.each do |v|
+            define_singleton_method(v) do
+              val = instance_variable_get "@#{v.to_s}".to_sym
+              raise "#{v} not set" if val.blank?
```

Comment:
will this always be true for all secrets?  is it guaranteed that blank will always be bad?

Reply:
Instead can you move this into the overridden `attr_accessor` method above?

```ruby
def self.attr_accessor(*vars, required: false)
  super(*vars)
  vars.each do |v|
    define_singleton_method(v) do
      val = instance_variable_get "@#{v.to_s}".to_sym
      raise "#{v} not set" if val.blank? && required
      val
    end
  end
end
```

Then below you can pick and choose which are required (they may all be now, but maybe not later):

```ruby
class Strategy1 < ConfigFields
  ...
  attr_accessor :secret_key, required: true
  ...
end